### PR TITLE
Progress on RB Post Photo Uploader updates.

### PR DESCRIPTION
**Pull request by weerd**
### What does this PR do?
This PR does a bunch of updates to the Reportback Post Photo Uploader to address some new visual design updates and provide some bug fixes.

### Any background context you want to provide?
Sure. There's been a new design for the RB uploader and while some of the ideas have been implemented, this PR adds the rest and does a bit of cleanup.

### Responsive Sizes
Here's what the new design looks like across the different responsive sizes:

#### Small
screen shot 2018-01-02 at 8 43 47 am

#### Medium
screen shot 2018-01-02 at 8 44 26 am

#### Large
screen shot 2018-01-02 at 8 44 53 am

### Image Sizes In Uploader
And it also account for any sized images that are uploaded, not just square ones:
screen shot 2018-01-02 at 8 45 55 am

screen shot 2018-01-02 at 8 46 22 am

screen shot 2018-01-02 at 8 46 43 am

### Validations
And here it is with the field validations as setup by @itsjoekent but with some design updates:
screen shot 2018-01-04 at 9 37 14 am

_Note: I did find a bug with validation where now since we allow campaign leads to toggle showing the impact field and we default it to `1` if it's toggled off (since it's also required by Rogue at the moment) we never really validate the impact field since it will always have a value. I feel we can ignore this for now and reassess, especially once the switch to posting to Rogue happens and quantity on the post is a thing.

### What are the relevant tickets/cards?
Refs https://www.pivotaltracker.com/story/show/153757088

### Checklist
- [x] Added appropriate feature/unit tests.
- [x] Added screenshot to PR description of related front-end updates on **small** screens.
- [x] Added screenshot to PR description of related front-end updates on **medium** screens.
- [x] Added screenshot to PR description of related front-end updates on **large** screens.

### CODE REVIEW

**Comment by weerd** [comment]
I'll be doing a further update in a following PR to include `flex`-ness in the form sections so the textarea (the why participated) extends as much as possible to fill the area :)

**Comment by weerd** [comment]
Some things to note. This really only aimed to tackle the design and bug fix updates! There's still an aim to refactor the component and simplify things in general when we start posting to Rogue but this gets us a step in the right direction. 💡

File: resources/assets/components/ReportbackUploader/ReportbackUploader.js [hunk 1]
```diff
@@ -105,13 +106,14 @@ class ReportbackUploader extends React.Component {
...
 
-    const shouldDisplaySubmissionMessaging = submissions.messaging && submissions.messaging.error;
+    const formHasErrors = get(submissions.messaging, 'error', null);
 
     const isInvalidField = name => (
       Object.keys(get(submissions, 'messaging.error.fields', {}))
         .indexOf(name) !== -1
     );
 
+    // @TODO: passing a hardcoded array is not sustainable...
```
Comment (itsjoekent) [inline resources/assets/components/ReportbackUploader/ReportbackUploader.js]: We will deal with it when we get to that point? doesn't seem like a big deal to me right now, was just an easy way to make the code dry

Comment (weerd) [reply resources/assets/components/ReportbackUploader/ReportbackUploader.js]: Yeah, thus the comment :) But we'll be dealing with it potentially soon if we have the refer a friend stuff coming in, etc.

**Pull request merged**